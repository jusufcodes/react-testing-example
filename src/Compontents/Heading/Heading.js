import React from "react";

export const Heading = (props) => {
  const level = props.level ? props.level : 1;

  const Elem = `h${level}`;

  return <Elem>{props.children}</Elem>
};

export const Avatar = (props) => {
  return props.src ? <img src={props.src} alt={props.alt} /> : null;
};