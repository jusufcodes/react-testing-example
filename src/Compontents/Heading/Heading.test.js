import React from 'react';
import { shallow } from 'enzyme';
import {Avatar, Heading} from "./Heading";
import renderer from 'react-test-renderer';

describe("Heading component", () => {

  test("Heading level by props", () => {
    for (let i = 1; i <= 6; i++) {
      const heading = shallow(<Heading level={i}>Welcome to React</Heading>);

      expect(heading.is(`h${i}`)).toEqual(true);
    }
  });

  test("Heading level without props", () => {
    const heading = shallow(<Heading>Welcome to React</Heading>);

    expect(heading.is("h1")).toEqual(true);
  });

  test("Heading text", () => {
    const heading = shallow(<Heading>Welcome to React</Heading>);

    expect(heading.text()).toEqual("Welcome to React");
  });

  test("Avatar without source", () => {
    const altText = "Placeholder Iamge";
    const avatar = <Avatar alt={altText} />;

    expect(shallow(avatar).isEmptyRender()).toEqual(true);

    const avatarSnapshot = renderer.create(avatar).toJSON();
    expect(avatarSnapshot).toMatchSnapshot();
  });

});
