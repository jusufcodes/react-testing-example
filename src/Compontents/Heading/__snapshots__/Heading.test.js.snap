// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Avatar without source 1`] = `null`;
